hack/generator/pkg/jsonast/jsonast.go

```diff
@@ -51,6 +51,9 @@ type (
 // findTypeDefinition looks to see if we have seen the specified definition before, returning its definition if we have.
 func (scanner *SchemaScanner) findTypeDefinition(name *astmodel.TypeName) (astmodel.TypeDefiner, bool) {
 	result, ok := scanner.definitions[*name]
+	if !ok || result == nil {
+		result, ok = scanner.definitions[*name.Singular()]
```

Comment:
I don’t much like this fallback, I’d rather wrap the call to `findTypeDefinition` with a check for `isResource`

Reply:
e.g. something like this around line 466; it will also get rid of the extra removal added on 507

```golang
if isResource {
    typeName = toResourceTypeName(typeName)
}
```

Reply:
I'll give that a try.

Reply:
Done; it's cleaner, and it works too. #alwaysimportant